File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        res=[]
        hashmap=defaultdict(int)
        for i in range(len(strs)):
            anares=''.join(sorted(strs[i]))
            if anares not in hashmap:
                hashmap[anares]=[]
            hashmap[anares].append(i)
        for key,value in hashmap.items():
            res.append([strs[val] for val in value])
        return res

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> res = new ArrayList<>();
        HashMap<String,List<String>> anagram_map = new HashMap<>();
        

        for(String str:strs){
            int[] temp = new int[26];
            for(char c:str.toCharArray()){
                temp[c - 'a']+=1;
            }
            String key=Arrays.toString(temp);
            //anagram_map.put(temp,str);
            anagram_map.putIfAbsent(key, new ArrayList<>());
            anagram_map.get(key).add(str);
        }
        anagram_map.forEach((key,value)->{
            res.add(value);
        });

        System.out.println(res.toString());
        System.out.println(anagram_map.toString());
        
        return res;

    }
}